src/main/java/com/aws/greengrass/deployment/DeploymentService.java

```diff
@@ -344,17 +353,23 @@ private void loadDeploymentQueueFromConfig() {
                 return;
             }
             logger.atInfo().kv(DEPLOYMENT_QUEUE_TOPIC, savedDeployments).log("Loading queued deployments");
+            final List<Deployment> deserializedDeployments = new ArrayList<>();
             savedDeployments.forEach(deploymentString -> {
                 try {
                     final Deployment deployment = SerializerFactory.getFailSafeJsonObjectMapper()
                             .readValue(deploymentString, Deployment.class);
                     if (deployment != null) {
-                        this.deploymentQueue.offer(deployment);
+                        deserializedDeployments.add(deployment);
                     }
                 } catch (JsonProcessingException e) {
                     logger.atError().cause(e).log("Failed to parse saved deployment queue element");
                 }
             });
+            // do not load deployments that target the nucleus itself, as doing so can cause undesired behavior.
+            deserializedDeployments.removeIf(this::doesDeploymentChangeTheNucleusVersion);
+            deserializedDeployments.forEach(deployment -> {
+                this.deploymentQueue.offer(deployment);
```

Comment:
this should also be a method reference I think

Reply:
done